docs/layouts/shortcodes/generated/dynamic_section.html

```diff
@@ -8,6 +8,12 @@
         </tr>
     </thead>
     <tbody>
+        <tr>
+            <td><h5>kubernetes.operator.argument.variables.enabled</h5></td>
+            <td style="word-wrap: break-word;">false</td>
+            <td>Boolean</td>
```

Comment:
Not sure what would be a good name but the feature is job args env var resolving, right? If one takes a look purely at the conf name then not able to tell the main intention. It would be good to give a name which reflects resolving. Maybe 'kubernetes.operator.job.arguments.env-var.resolve.enabled' or something.

Reply:
`kubernetes.operator.flink-config.env-variables.enabled` or something like that

Reply:
ah okay, I was under the impression that this PR worked for all flink configs

Reply:
This is already given by `kubernetes.jobmanager.entrypoint.args`, the extra that we're adding here is to add the internal args automatically, avoid users manually dealing with the property`-D\$internal.application.program-args=--sleep\;${SLEEP}\;--error-rate\;${ERROR_RATE}`

Reply:
It's somewhat of a hidden feature technically. I thought instead of just applying the current logic to job args we could apply it to all Flink Config key/value pairs.

That would cover both args and all else too.

Reply:
With `kubernetes.jobmanager.entrypoint.args` you already getting this functionality you're requesting. If you think it's sufficient we can simply close out this one.

Reply:
I think it's nice if the operator provides these convenience features, so the user doesn't need to figure it out how to do this.

So I completely agree with this PR, all I am trying to say is that we could extend the scope to the entire spec.flinkConfiguration not only to spec.job.args

Reply:
but we don't have to do this right now, we can also add this later

Reply:
Ah ok I guess I see your point. So you want automatic substitution of:
```
security.ssl.internal.truststore: ${TRUSTSTORE}
security.ssl.internal.truststore-password: ${TRUSTSTORE_PASSWROD}
```

Instead of:

```
kubernetes.jobmanager.entrypoint.args: -Dsecurity.ssl.internal.truststore: ${TRUSTSTORE} -Dsecurity.ssl.internal.truststore-password: ${TRUSTSTORE_PASSWROD}
```

I'd prefer to do this in a separate jira.

Reply:
The only problem then is that we will have to create a new config and immediately deprecate the one you just created now, because the other will completely subsume it. 

We could also just take a little more time and do this properly instead of rushing this in for the release. Users already have a workaround and I would prefer to get this right at first.

Reply:
ok, closing the pr for the time being